src/types.ts

```diff
@@ -1,6 +1,6 @@
 import Piece from './pieces/piece'
 
-export type Field = (Piece | null);
+export type Field = (Piece | undefined);
```

Comment:
Przeanalizowałem to troche i to nie bedzie takie trywialne niestety ten typ na board chyba, zamieńmy jak możesz type na interface żeby zachować spójność w apce, @KlaraGajaszek undefined?